scrypto/src/resource/resource_manager.rs

```diff
@@ -32,6 +31,51 @@ pub enum Mutability {
     MUTABLE(AccessRule),
 }
 
+#[derive(Debug, TypeId, Encode, Decode)]
+pub enum ResourceManagerFunction {
+    Create(
+        ResourceType,
+        HashMap<String, String>,
+        HashMap<ResourceMethod, (AccessRule, Mutability)>,
+        Option<MintParams>,
+    )
+}
+
+#[derive(Debug, TypeId, Encode, Decode)]
+pub enum ResourceManagerMethod {
+    Mint(MintParams),
+    UpdateAuth(ResourceMethod, AccessRule),
+    LockAuth(ResourceMethod),
+    GetResourceType(),
+    GetMetadata(),
+    GetTotalSupply(),
+    GetNonFungible(NonFungibleId),
+    NonFungibleExists(NonFungibleId),
+    UpdateNonFungibleData(NonFungibleId, Vec<u8>),
+    UpdateMetadata(HashMap<String, String>),
+    CreateVault(),
+    CreateBucket(),
+}
+
+impl ResourceManagerMethod {
+    pub fn name(&self) -> &str {
+        match self {
+            ResourceManagerMethod::Mint(_) => "mint",
+            ResourceManagerMethod::UpdateAuth(_, _) => "update_auth",
+            ResourceManagerMethod::LockAuth(_) => "lock_auth",
+            ResourceManagerMethod::GetResourceType() => "get_resource_type",
+            ResourceManagerMethod::GetMetadata() => "get_metadata",
+            ResourceManagerMethod::GetTotalSupply() => "get_total_supply",
+            ResourceManagerMethod::GetNonFungible(_) => "get_non_fungible",
+            ResourceManagerMethod::NonFungibleExists(_) => "non_fungible_exists",
+            ResourceManagerMethod::UpdateNonFungibleData(_, _) => "update_non_fungible_data",
+            ResourceManagerMethod::UpdateMetadata(_) => "update_metadata",
+            ResourceManagerMethod::CreateVault() => "create_vault",
+            ResourceManagerMethod::CreateBucket() => "create_bucket",
+        }
```

Comment:
Didn't get why we still need these string literals.

Reply:
This is required to be able to add the range of methods to maps. But agreed this needs more reworking.

Reply:
Sorry, what do you mean by "the range of methods"? Wildcard matching?